### ISSUE TITLE
Failure calling scripty MSBuild task from Visual Studio

### ISSUE THREAD

**Issue by OliverHallam** [daveaglick/Scripty#80]
I've updated our project to use the Scripty MSBuild Task.  This works fine when built with MSBuild on the command line, but it fails to invoke Scripty when calling from Visual Studio (both 2015 and 2017).

It looks like it's down to the logic that copies the global MSBuild properties (in `src/Scripty.MsBuild/ScriptyTask.cs`)

```c#
            foreach (var property in properties)
            {
                args.Add($"--p \"{property.Key}={property.Value}\"");
            }
```

In this case we are not doing any escaping on the property value.

I'm seeing visual studio pass a big block of XML (including newlines and quote characters) for the `CurrentSolutionConfigurationContents` property which is being misinterpreted by Scripty.exe:

```
error : Got non-zero exit code from script evaluation: 1
error : System.ArgumentException: Illegal characters in path.
error :    at System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
error :    at System.IO.Path.Combine(String path1, String path2)
error :    at Scripty.Program.Run(String[] args)
error :    at Scripty.Program.Main(String[] args)
```

**Comment by OliverHallam** [comment]
It looks like this is related to #66

**Comment by CZEMacLeod** [comment]
Yup - I've got the same problem here - Illegal characters in path.
I hadn't managed to get much investigation done on it though.

**Comment by OliverHallam** [comment]
As a short term workaround, we're calling Scripty.exe directly - it's not as nice; you need to add the generated files to the project manually - but it serves our needs for now.

In case anyone else is running against this, we've created our own .targets file based on the one in the MSBuild package, but replaced the final target with this:

```xml
  <PropertyGroup>
    <ScriptyExecutable Condition=" '$(ScriptyExecutable)' == '' ">$(MSBuildThisFileDirectory)\..\Tools\Scripty\Scripty.exe</ScriptyExecutable>
  </PropertyGroup>

  <Target Name="EvaluateScriptyFiles" DependsOnTargets="_ResolveScriptyFiles" BeforeTargets="BeforeBuild">
    <Exec Command="$(ScriptyExecutable) --solution $(SolutionPath) $(MSBuildProjectFullPath) @(ScriptyFile -> '%(FullPath)', ' ')" />
  </Target>
```

**Comment by daveaglick** [comment]
@OliverHallam Finally starting to take a look at the issue backlog now that I've got the new version out. I'm having some trouble replicating this - any ideas on how to generate that big `CurrentSolutionConfigurationContents` block? I'll add escaping for the build properties, but it would be nice to repro this first.

**Comment by daveaglick** [comment]
Going to solve this and related problems by sticking all the data into JSON and sending via stdin, bypassing the CLI entirely

**Comment by daveaglick** [comment]
This should be resolved in the next release, 0.7.2
